packages/devtools_app/lib/src/standalone_ui/vs_code/flutter_panel.dart

```diff
@@ -2,18 +2,119 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 
+import '../../../devtools_app.dart';
 import '../../shared/feature_flags.dart';
+import '../api/dart_tooling_api.dart';
+import '../api/vs_code_api.dart';
+
+/// A general Flutter sidebar panel for embedding inside IDEs.
+///
+/// Provides some basic functionality to improve discoverability of features
+/// such as creation of new projects, device selection and DevTools features.
+class VsCodeFlutterPanel extends StatefulWidget {
+  const VsCodeFlutterPanel(this.api, {super.key});
+
+  final DartToolingApi api;
 
-class VsCodeFlutterPanel extends StatelessWidget {
-  const VsCodeFlutterPanel({super.key});
+  @override
+  State<VsCodeFlutterPanel> createState() => _VsCodeFlutterPanelState();
+}
 
+class _VsCodeFlutterPanelState extends State<VsCodeFlutterPanel> {
   @override
   Widget build(BuildContext context) {
     assert(FeatureFlags.vsCodeSidebarTooling);
-    return const Center(
-      child: Text('TODO: a panel for flutter actions in VS Code'),
+
+    final api = widget.api;
+
+    return Column(
+      children: [
+        const Text(''),
```

Comment:
I replaced this with `const SizedBox(height: 20),` (it was to add some padding to prevent the first item in the column touching the top of the window), but I'm not sure if that's the best way to do this?